### PR TITLE
[NEXUS-8556] avoid checking repository status while Nexus is still starting up

### PR DESCRIPTION
**Pull request by mcculls**
https://issues.sonatype.org/browse/NEXUS-8556
http://bamboo.s/browse/NX-OSSF820-1 :white_check_mark: 

Whenever a proxy repository is loaded/created it spawns a RepositoryStatusCheckerThread to perform various checks. To avoid false-positives we should hold them back from checking until Nexus has started and the configuration is fully loaded.

### CODE REVIEW

**Comment by mrprescott** [comment]
+1

**Comment by adreghiciu** [comment]
+1

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/RepositoryStatusCheckerThread.java [hunk 1]
```diff
@@ -53,6 +60,18 @@ public void run() {
     boolean interrupted = false;
 
     while (isRunning() && getRepository().getProxyMode() != null) {
+
+      // wait for Nexus to fully start
+      if (!systemStatus.isNexusStarted()) {
+        try {
+          Thread.sleep(10 * 1000);
```
Comment (kellyrob99) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/RepositoryStatusCheckerThread.java]: Given that the isNexusStarted() check seems very lightweight, might want to set a smaller interval here?

**Comment by kellyrob99** [comment]
+1

File: components/nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/RepositoryStatusCheckerThread.java (hunk 1 above)
Comment (mcculls) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/RepositoryStatusCheckerThread.java]: Each proxy repo has one of these threads, so for large configurations a very small interval here could be counter-productive. Also the main interval of these checker threads once started is typically set to 1 hour, so delaying them post-startup by at most 10s is not a big deal in the overall context of what they're doing. Were you thinking of 1s, 3s, 5s?

Comment (kellyrob99) [inline components/nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/RepositoryStatusCheckerThread.java]: the smaller the better, but your explanation seems to make it a moot point so nm

**Comment by mcculls** [comment]
Merged manually under 0903225

**Pull request closed without merging**